presto-main/src/test/java/com/facebook/presto/type/TestRowOperators.java

```diff
@@ -169,5 +187,35 @@ public void testRowEquality()
         assertFunction("test_row(TRUE, ARRAY [1]) != test_row(TRUE, ARRAY [1])", BOOLEAN, false);
         assertFunction("test_row(TRUE, ARRAY [1]) != test_row(TRUE, ARRAY [1,2])", BOOLEAN, true);
         assertFunction("test_row(1.0, ARRAY [1,2,3], test_row(2,2.0)) != test_row(1.0, ARRAY [1,2,3], test_row(1,2.0))", BOOLEAN, true);
+
+        // Cannot be applied with different field names
+        // assertFunction("CAST(ROW(1, 2) AS ROW(a BIGINT, b BIGINT)) = CAST(ROW(1, 2) AS ROW(b BIGINT, a BIGINT))", BOOLEAN, true);
+        // assertFunction("CAST(ROW(1, 2) AS ROW(a BIGINT, b BIGINT)) != CAST(ROW(2, 1) AS ROW(b BIGINT, a BIGINT))", BOOLEAN, true);
+        // assertFunction("ROW(1, 2) = CAST(ROW(1, 2) AS ROW(a BIGINT, b BIGINT))", BOOLEAN, true);
+        // assertFunction("CAST(ROW(1, 2) AS ROW(a BIGINT, b BIGINT)) = ROW(1, 2)", BOOLEAN, true);
+        assertFunction("ROW(1, 2) = ROW(1, 2)", BOOLEAN, true);
+        assertFunction("ROW(2, 1) != ROW(1, 2)", BOOLEAN, true);
+    }
+
+    @Test
+    public void testRowSelect()
+            throws Exception
+    {
+        testQuery("SELECT ROW(1 + 2, CONCAT('a', 'b'))", "SELECT ROW(3, 'ab')");
```

Comment:
Add these tests to `AbstractTestQueries` instead of here. Then you can remove the `queryRunner`